python/pyspark/ml/torch/tests/test_distributor.py

```diff
@@ -133,6 +214,42 @@ def test_get_num_tasks_locally(self) -> None:
                     distributor = TorchDistributor(num_processes, True, True)
                     distributor.num_processes = 3
 
+    def test_get_gpus_owned_local(self) -> None:
+        addresses = ["0", "1", "2"]
+        self.assertEqual(get_gpus_owned(self.sc), addresses)
+
+        env_vars = {"CUDA_VISIBLE_DEVICES": "3,4,5"}
+        self.setup_env_vars(env_vars)
+        self.assertEqual(get_gpus_owned(self.sc), ["3", "4", "5"])
+        self.delete_env_vars(env_vars)
+
+    def test_local_training_succeeds(self) -> None:
```

Comment:
We have a baseline test here, but there will need to be an integration test down the line (https://issues.apache.org/jira/browse/SPARK-41777)